corehq/apps/locations/models.py

```diff
@@ -436,19 +436,27 @@ def delete(self, *args, **kwargs):
     full_delete = delete
 
     def get_descendants(self, include_self=False, **kwargs):
+        include_archived = kwargs.pop('include_archived', True)
         if include_self:
             where = Q(domain=self.domain, id=self.id)
         else:
             where = Q(domain=self.domain, parent_id=self.id)
-        return SQLLocation.objects.get_descendants(
-            where, **kwargs
-        )
+        if include_archived:
+            locations = SQLLocation.objects
+        else:
+            locations = SQLLocation.active_objects
```

Comment:
There's no need to switch model managers here - all that does is control whether a filter is applied automatically.  Since it's not being applied automatically, IMO it'd be more direct to apply the filter explicitly:
```python
queryset = SQLLocation.objects.get_descendants(where, **kwargs)
if not include_archived:
    queryset = res.filter(is_archived=False)
return queryset
```

Reply:
okay, i used them because they already internally handle that distinction so thought it was better reuse what is already available.

Reply:
Yes, exactly!  That's the pattern that was sortof implemented here, but whoever did it (I hope it wasn't me) was probably nervous about mandating that change across the board and punted the decision to the calling code, leaving us in this weird situation where it's not obvious what the intent was.

Reply:
As an aside - I think one way to approach changing that system-wide would be to add to the class:
```python
    all_objects = objects  # includes even archived locations
```
Then we can slowly switch all references to `SQLLocation.objects` to either `SQLLocation.all_objects` or `SQLLocation.active_objects`.  Then when there are no more references to `SQLLocation.objects`, we can bulk rename `active_objects` to `objects`.  I wouldn't bother unless doing a lot of work in the area anyways, though.